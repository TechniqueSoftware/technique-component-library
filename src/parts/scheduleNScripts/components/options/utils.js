export const followUpTimeToObject = followUpTime => {
  const getHour = h => {
    if (h === '00') return 12
    const parsedHour = parseInt(h, 10)
    return parsedHour > 12 ? parsedHour - 12 : parsedHour
  }
  const time = followUpTime.split(':')
  const [oldHour, oldMinute] = time
  const hour = getHour(oldHour)
  const minute = parseInt(oldMinute, 10)
  const meridiem = oldHour >= 12 ? 'PM' : 'AM'
  return { hour, minute, meridiem }
}

export const objectToFollowUpTime = (hour, minute, meridiem) => {
  const parseHour = h => meridiem === 'PM' ? parseInt(h, 10) + 12 : h
  return `${parseHour(hour === 12 ? '00' : hour)}:${minute === 0 ? '00' : minute}:00`
}
