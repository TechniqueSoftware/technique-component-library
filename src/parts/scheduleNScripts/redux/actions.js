import TYPES from './types'

const FOLLOWUPS_ACTIONS = {
  GET: () => ({ type: TYPES.FOLLOWUPS.GET }),
  ERROR: payload => ({ type: TYPES.FOLLOWUPS.ERROR, payload }),
  SUCCESS: payload => ({ type: TYPES.FOLLOWUPS.SUCCESS, payload }),
  PERMISSION: payload => ({ type: TYPES.FOLLOWUPS.PERMISSION, payload }),
  SET: payload => ({ type: TYPES.FOLLOWUPS.SET, payload }),
  INNER_PERMISSIONS: payload => ({ type: TYPES.FOLLOWUPS.INNER_PERMISSIONS, payload }),
  SET_INNER_PERMISSIONS: payload => ({ type: TYPES.FOLLOWUPS.SET_INNER_PERMISSIONS, payload }),
  GET_FEATURE_PERMISSIONS: payload => ({ type: TYPES.FOLLOWUPS.GET_FEATURE_PERMISSIONS, payload })
}

const TOUCHPOINT_ACTIONS = {
  OPEN: payload => ({ type: TYPES.TOUCHPOINT.OPEN, payload }),
  EDIT: payload => ({ type: TYPES.TOUCHPOINT.EDIT, payload }),
  EDIT_TOUCHPOINT: payload => ({ type: TYPES.TOUCHPOINT.EDIT_TOUCHPOINT, payload }),
  EDIT_SUCCESS: payload => ({ type: TYPES.TOUCHPOINT.EDIT_SUCCESS, payload }),
  EDIT_ERROR: payload => ({ type: TYPES.TOUCHPOINT.EDIT_ERROR, payload }),
  NEW: payload => ({ type: TYPES.TOUCHPOINT.NEW, payload }),
  NEW_SUCCESS: payload => ({ type: TYPES.TOUCHPOINT.NEW_SUCCESS, payload }),
  NEW_ERROR: payload => ({ type: TYPES.TOUCHPOINT.NEW_ERROR, payload }),
  DELETE: payload => ({ type: TYPES.TOUCHPOINT.DELETE, payload }),
  DELETE_SUCCESS: payload => ({ type: TYPES.TOUCHPOINT.DELETE_SUCCESS, payload }),
  DELETE_ERROR: payload => ({ type: TYPES.TOUCHPOINT.DELETE_ERROR, payload }),
  CLOSE: () => ({ type: TYPES.TOUCHPOINT.CLOSE })
}

const SCRIPT_ACTIONS = {
  GET: () => ({ type: TYPES.SCRIPTS.GET }),
  SAVE: payload => ({ type: TYPES.SCRIPTS.SAVE, payload }),
  EDIT: payload => ({ type: TYPES.SCRIPTS.EDIT, payload })
}

const SEGMENTS_ACTIONS = {
  CLOSE: () => ({ type: TYPES.SEGMENTS.CLOSE }),
  GET: payload => ({ type: TYPES.SEGMENTS.GET, payload }),
  SET: payload => ({ type: TYPES.SEGMENTS.SET, payload }),
  GET_ATTRIBUTES: () => ({ type: TYPES.SEGMENTS.GET_ATTRIBUTES }),
  SET_ATTRIBUTES: payload => ({ type: TYPES.SEGMENTS.SET_ATTRIBUTES, payload }),
  SET_DATA: payload => (({ type: TYPES.SEGMENTS.SET_DATA, payload })),
  SAVE: payload => ({ type: TYPES.SEGMENTS.SAVE, payload }),
  DELETE: payload => ({ type: TYPES.SEGMENTS.DELETE, payload }),
  GET_LOCATIONS: () => ({ type: TYPES.SEGMENTS.GET_LOCATIONS }),
  SET_LOCATIONS: payload => ({ type: TYPES.SEGMENTS.SET_LOCATIONS, payload }),
  GET_LIST: payload => ({ type: TYPES.SEGMENTS.GET_LIST, payload }),
  SET_LIST: payload => ({ type: TYPES.SEGMENTS.SET_LIST, payload }),
  GET_COPY_PROCESS: payload => ({ type: TYPES.SEGMENTS.GET_COPY_PROCESS, payload }),
  SET_COPY_PROCESS: payload => ({ type: TYPES.SEGMENTS.SET_COPY_PROCESS, payload }),
  SET_ALERT: payload => ({ type: TYPES.SEGMENTS.SET_ALERT, payload }),
  ERROR: payload => ({ type: TYPES.SEGMENTS.ERROR, payload })
}

const SNACKBAR_ACTIONS = {
  CLOSE: () => ({ type: TYPES.SNACKBAR.CLOSE }),
  SET: payload => ({ type: TYPES.SNACKBAR.SET, payload })
}

export default {
  FOLLOWUPS: FOLLOWUPS_ACTIONS,
  TOUCHPOINT: TOUCHPOINT_ACTIONS,
  SCRIPTS: SCRIPT_ACTIONS,
  SEGMENTS: SEGMENTS_ACTIONS,
  SNACKBAR: SNACKBAR_ACTIONS
}
