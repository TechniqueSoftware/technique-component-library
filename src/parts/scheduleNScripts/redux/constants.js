export const TP_OPTIONS = {
  MENU: 'menu',
  NEW: 'new',
  EDIT: 'edit'
}

export const GROUP_NAMES = {
  PROSPECT: 'Prospect',
  ACTIVE_MEMBER: 'Active Member',
  INACTIVE_MEMBER: 'Inactive Member',
  PT_PROSPECT: 'PT Prospect',
  PT_MEMBER: 'PT Member',
  PT_OPT_OUT: 'PT Opt Out',
  PERSONAL_TRAINING: 'Personal Training'
}
