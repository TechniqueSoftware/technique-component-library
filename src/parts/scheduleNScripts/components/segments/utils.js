import { FOLLOWUP_STATUS } from './constants'

export const parseTitle = followUpStatus => {
  switch (followUpStatus) {
    case FOLLOWUP_STATUS.WEB_LEAD:
      return 'Web Lead'
    case FOLLOWUP_STATUS.APPT_BOOKED:
      return 'Appt Booked'
    case FOLLOWUP_STATUS.MEMBER:
      return 'Member'
  }
}

export const parseLocations = segmentLocations => segmentLocations.map(({ locationName, locationId }) => ({
  label: `${locationName}`,
  value: locationId
}))

export const checkPermission = userPermission => userPermission.some(
  element => element.permissionName === 'CopySegmentsAction' && element.permissionEffect === 'ALLOW')
