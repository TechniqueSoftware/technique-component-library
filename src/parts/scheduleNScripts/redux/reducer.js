import { produce } from 'immer'
import { TP_OPTIONS } from './constants'
import TYPES from './types'

export const initialState = {
  loading: false,
  error: '',
  snackbar: '',
  followups: [],
  canEditScripts: false,
  canEditSchedules: false,
  touchpointCrud: {
    action: '',
    data: {},
    loading: false
  },
  script: {},
  segments: [],
  segmentModalData: {
    isOpen: false,
    title: '',
    error: ''
  },
  segmentAttributes: {
    marketing: [],
    membership: []
  },
  innerPermissions: {
    automaticFollowUpAction: false,
    followUpSegmentsAction: false
  },
  segmentLocations: [],
  segmentList: [],
  copyInProcess: {},
  popupAlert: ''
}

const reducer = (state = initialState, { type, payload }) => produce(state, draft => {
  switch (type) {
    case TYPES.FOLLOWUPS.GET:
      draft.loading = true
      draft.followups = []
      draft.error = ''
      break

    case TYPES.FOLLOWUPS.ERROR:
      draft.loading = false
      draft.error = payload
      draft.followups = []
      break

    case TYPES.FOLLOWUPS.SUCCESS:
      draft.loading = false
      draft.error = ''
      draft.followups = payload.followups
      draft.canEditSchedules = payload.canEditSchedules
      draft.canEditScripts = payload.canEditScripts
      break

    case TYPES.FOLLOWUPS.PERMISSION:
      draft.canEditSchedules = payload.canEditSchedules
      draft.canEditScripts = payload.canEditScripts
      break

    case TYPES.FOLLOWUPS.SET:
      draft.followups = payload
      break

    case TYPES.FOLLOWUPS.SET_INNER_PERMISSIONS:
      draft.innerPermissions = payload
      break

    case TYPES.TOUCHPOINT.OPEN:
      draft.touchpointCrud = {
        action: payload.action,
        data: payload.data
      }
      break

    case TYPES.TOUCHPOINT.CLOSE:
      draft.touchpointCrud = { action: '', data: {}, loading: false }
      break

    case TYPES.TOUCHPOINT.NEW:
      draft.touchpointCrud.loading = true
      break

    case TYPES.TOUCHPOINT.NEW_SUCCESS:
      draft.followups = payload
      draft.touchpointCrud.loading = false
      draft.snackbar = 'NEW_SUCCESS'
      break

    case TYPES.TOUCHPOINT.NEW_ERROR:
      draft.touchpointCrud.loading = false
      draft.touchpointCrud.error = payload
      break

    case TYPES.TOUCHPOINT.DELETE:
      draft.touchpointCrud.loading = true
      break

    case TYPES.TOUCHPOINT.DELETE_SUCCESS:
      draft.followups = payload
      draft.touchpointCrud.loading = false
      draft.snackbar = 'DELETE_SUCCESS'
      break

    case TYPES.TOUCHPOINT.DELETE_ERROR:
      draft.touchpointCrud.loading = false
      draft.touchpointCrud.error = payload
      break

    case TYPES.TOUCHPOINT.EDIT:
      draft.touchpointCrud.action = TP_OPTIONS.EDIT
      break

    case TYPES.TOUCHPOINT.EDIT_TOUCHPOINT:
      draft.touchpointCrud.loading = true
      break

    case TYPES.TOUCHPOINT.EDIT_SUCCESS:
      draft.followups = payload.followups
      draft.touchpointCrud.loading = false
      draft.snackbar = payload.snackbar
      break

    case TYPES.TOUCHPOINT.EDIT_ERROR:
      draft.touchpointCrud.loading = false
      draft.touchpointCrud.error = payload
      break

    case TYPES.SNACKBAR.CLOSE:
      draft.snackbar = ''
      break

    case TYPES.SNACKBAR.SET:
      draft.snackbar = payload
      break

    case TYPES.SCRIPTS.EDIT:
      draft.touchpointCrud = {
        action: payload.action,
        data: payload.data
      }
      break

    case TYPES.SCRIPTS.SAVE:
      draft.script = {
        data: payload.data
      }
      break

    case TYPES.SEGMENTS.SET:
      draft.segments = payload
      break

    case TYPES.SEGMENTS.SET_DATA:
      draft.segmentModalData.title = payload.title
      draft.segmentModalData.isOpen = true
      draft.segmentModalData.followUpType = payload.followUpType
      draft.segmentModalData.followUpStatus = payload.followUpStatus
      break

    case TYPES.SEGMENTS.SET_ATTRIBUTES:
      draft.segmentAttributes = payload
      break

    case TYPES.SEGMENTS.CLOSE:
      draft.segmentModalData.isOpen = false
      draft.segmentModalData.error = ''
      break

    case TYPES.SEGMENTS.ERROR:
      draft.segmentModalData.error = payload
      break

    case TYPES.SEGMENTS.SET_LOCATIONS:
      draft.segmentLocations = payload
      break

    case TYPES.SEGMENTS.SET_LIST:
      draft.segmentList = payload
      break

    case TYPES.SEGMENTS.SET_ALERT:
      draft.popupAlert = payload
      break

    case TYPES.SEGMENTS.SET_COPY_PROCESS:
      draft.copyInProcess = payload
      break
  }
})

export default reducer
