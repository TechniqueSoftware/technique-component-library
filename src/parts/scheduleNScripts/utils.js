import { PERMISSION_KEYS, STATUSES_WITH_SEGMENTS } from './constants'

export const checkIfSegmentStatus = (type, id) => {
  const hasSegmentation = !!STATUSES_WITH_SEGMENTS.find(s => s.type === parseInt(type, 10) && s.id === parseInt(id, 10))
  return hasSegmentation
}

const setSchedulesDayNum = schedules => schedules.reduce(
  (acc, curr) => {
    const dayNum = curr.daysTillNext + acc.dayNum
    return { schedules: acc.schedules.concat({ ...curr, dayNum }), dayNum }
  },
  { dayNum: 0, schedules: [] }
).schedules

export const getStatusSegments = (type, id, segments) => {
  if (!checkIfSegmentStatus(type, id)) return []
  const currentStatus = segments.find(
    ({ followUpType, followUpStatus }) => followUpType === parseInt(type, 10) && followUpStatus === parseInt(id, 10)
  )
  if (!currentStatus) return []
  const { followUpSegments = [], followUpSchedules = [], followUpSegmentScripts = [] } = currentStatus
  return followUpSegments.map(segment => {
    const schedulesWithScripts = followUpSchedules.reduce((acc, schedule) => {
      if (segment.segmentId !== schedule.segmentId) return acc
      return acc.concat({
        ...schedule,
        scripts: followUpSegmentScripts.filter(
          script => script.followUpSeqNum === schedule.followUpSeqNum && script.segmentId === schedule.segmentId
        )
      })
    }, [])
    return { ...segment, followUpSchedules: setSchedulesDayNum(schedulesWithScripts) }
  })
}

export const sortSegments = segments => [...segments].sort((a, b) => a.name.localeCompare(b.name))

// Formatting club-os data into something usable
export const parseSnSData = (data, followUpSchedules, followUpScripts, segmentsData, isClub) => {
  const schedulesPermissions = followUpSchedules.permissions
  const scriptsPermissions = followUpScripts.permissions
  const schedules = isClub ? followUpSchedules : followUpSchedules.schedules
  const scripts = isClub ? followUpScripts : followUpScripts.scripts

  const group = (acc, curr) => {
    const { followUpStatus, followUpType } = curr
    const key = `${followUpStatus}-${followUpType}`
    if (acc[key]) acc[key].push(curr)
    else acc[key] = [curr]
    return acc
  }

  const sch = schedules.reduce(group, {})
  const scr = scripts.reduce(group, {})

  Object.keys(sch).forEach(key => { sch[key] = setSchedulesDayNum(sch[key]) })

  Object.keys(scr).forEach(key => {
    const scripts = scr[key]
    scripts.forEach(script => {
      const schedule = sch[key]?.find(el => el.followUpSeqNum === script.followUpSeqNum)
      if (!schedule) return
      if (schedule.scripts) schedule.scripts.push(script)
      else schedule.scripts = [script]
    })
  })

  const followups = data.map(el => {
    const { type, statuses, id } = el
    const canEditSchedules = isClub ? undefined : schedulesPermissions[PERMISSION_KEYS.SCHEDULES[type]]
    const canEditScripts = isClub ? undefined : scriptsPermissions[PERMISSION_KEYS.SCRIPTS[type]]
    const newStatuses = statuses.map(st => {
      const key = `${st.id}-${type}`
      const unsortedSegments = getStatusSegments(type, st.id, segmentsData)
      const segments = sortSegments(unsortedSegments)
      return { ...st, followUpTab: id, schedules: sch[key] || [], segments }
    })
    return { ...el, statuses: newStatuses, canEditSchedules, canEditScripts }
  })

  return {
    followups,
    canEditSchedules: followups[0].canEditSchedules,
    canEditScripts: followups[0].canEditScripts
  }
}

export const getSchedulesByDay = (schedules, dayNum, order) => {
  const { earlierSch, sameDaySch, laterSch } = schedules.reduce(
    (acc, sch) => {
      if (sch.dayNum < dayNum) acc.earlierSch = acc.earlierSch.concat(sch)
      if (sch.dayNum === dayNum) acc.sameDaySch = acc.sameDaySch.concat(sch)
      if (sch.dayNum > dayNum) acc.laterSch = acc.laterSch.concat(sch)
      return acc
    },
    { earlierSch: [], sameDaySch: [], laterSch: [] }
  )
  return {
    earlierSch: order === 'first' ? earlierSch : earlierSch.concat(sameDaySch),
    laterSch: order === 'first' ? sameDaySch.concat(laterSch) : laterSch
  }
}

export const getCurrentSchedules = (segmentId, status) => {
  if (!segmentId) return status.schedules
  return status.segments.find(segment => segment.segmentId === segmentId).followUpSchedules
}

export const getStatus = (followups, type, status) => {
  const fs = followups.filter(f => f.type === type)
  if (!fs.length) return null

  let st = null
  for (let i = 0; i < fs.length; i++) {
    st = fs[i].statuses.find(s => s.id === `${status}`)
    if (st) break
  }
  return st
}

export const setFollowUpStatusValue = (followUps, status, value) => {
  const newFollowups = followUps.map(followUp => {
    if (followUp.id !== status.followUpTab) return followUp
    const newStatuses = followUp.statuses.map(st => {
      if (st.id !== status.id) return st
      return { ...st, ...value }
    })
    return { ...followUp, statuses: newStatuses }
  })
  return newFollowups
}

export const setFollowUpScripts = (followUps, { followUpType, followUpStatus }, scriptsData, segmentId) => {
  const status = getStatus(followUps, followUpType, followUpStatus)
  const currentSchedules = getCurrentSchedules(segmentId, status)
  const schedules = currentSchedules.map(schedule => {
    if (schedule.followUpSeqNum !== scriptsData.followUpSeqNum) return schedule
    return scriptsData
  })
  const statusValueToUpdate = !segmentId
    ? { schedules }
    : {
      segments: status.segments.map(
        segment => (segment.segmentId !== segmentId ? segment : { ...segment, followUpSchedules: schedules }) // eslint-disable-line max-len
      )
    }
  return setFollowUpStatusValue(followUps, status, statusValueToUpdate)
}

export const updateTouchpointInFollowups = (followUps, tp, data) => {
  const { followUpType, followUpStatus, followUpSeqNum } = tp
  const status = getStatus(followUps, followUpType, followUpStatus)
  const newSchedules = status.schedules.map(sch => {
    if (sch.followUpSeqNum !== followUpSeqNum) return sch
    const newSch = { ...sch, ...data }
    return newSch
  })
  return setFollowUpStatusValue(followUps, status, { schedules: newSchedules })
}
