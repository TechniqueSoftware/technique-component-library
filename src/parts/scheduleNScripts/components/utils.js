import { STATUSES_WITH_SEGMENTS } from '@parts/scheduleNScripts/constants'
// SEGMENTS
export const checkIfAttributeIsInUse = (segments, selectedAttributes, valueToCheck, segmentId) => {
  if (valueToCheck === '') return false
  const listToCheck = segments.reduce(
    (acc, segment) => {
      if (segmentId && segment.segmentId === segmentId) return acc
      return acc.concat(segment.segmentAttributes)
    },
    selectedAttributes.map(({ value }) => ({ attributeValue: value }))
  )
  return !!listToCheck.find(({ attributeValue }) => attributeValue === valueToCheck)
}

export const getSegmentStatusProp = ({ followUpType, followUpStatus }, keyword) => {
  const segmentTp = STATUSES_WITH_SEGMENTS.find(
    s => s.type === parseInt(followUpType, 10) && s.id === parseInt(followUpStatus, 10)
  )
  return segmentTp?.[keyword]
}

export const parseAttributes = (attributes, type) => attributes.reduce((acc, attr) => {
  if (!attr.value) return acc
  return acc.concat({ attributeValue: attr.value, attributeType: type, attributeId: attr.id })
}, [])

export const unparseAttributes = attributes => attributes.map(({ attributeValue, attributeId }) => ({
  name: attributeValue,
  value: attributeValue,
  id: attributeId
}))
