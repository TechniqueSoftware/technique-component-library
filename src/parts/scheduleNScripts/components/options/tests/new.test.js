import React from 'react'
import * as redux from 'react-redux'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { TestProvider as Provider } from '@redux/utils'

import { ACTIONS } from '../../../redux'

import New from '../new'

const mockDispatch = jest.fn()
const setHasCloseButton = jest.fn()
jest.spyOn(redux, 'useDispatch').mockReturnValue(mockDispatch)

describe('<New />', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const { asFragment } = render(
      <Provider>
        <New followUpSeqNum={1} dayNum={1} />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders options', () => {
    render(
      <Provider>
        <New />
      </Provider>
    )
    expect(screen.queryByText('Immediate')).toBeInTheDocument()
    expect(screen.queryByText('Day')).toBeInTheDocument()
    expect(screen.getByRole('spinbutton')).toBeInTheDocument()

    expect(screen.queryByText('Last')).not.toBeInTheDocument()
    expect(screen.queryByText('First')).not.toBeInTheDocument()
  })

  describe('creates new day', () => {
    it('creates new day', () => {
      render(
        <Provider>
          <New />
        </Provider>
      )

      userEvent.click(screen.queryByText('Day'))
      userEvent.type(screen.getByRole('spinbutton'), '12')
      userEvent.click(screen.getByText('Save'))

      expect(mockDispatch).toHaveBeenCalledWith(
        ACTIONS.TOUCHPOINT.NEW({
          dayNum: 12
        })
      )
    })

    it('shows options if same day exists', () => {
      render(
        <Provider>
          <New tpDays={['3']} setHasCloseButton={setHasCloseButton} />
        </Provider>
      )

      userEvent.click(screen.queryByText('Day'))
      userEvent.type(screen.getByRole('spinbutton'), '3')
      userEvent.click(screen.getByText('Save'))

      expect(mockDispatch).toHaveBeenCalled()

      expect(setHasCloseButton).toHaveBeenCalledTimes(1)

      expect(screen.getByText('Last')).toBeInTheDocument()
      expect(screen.getByText('First')).toBeInTheDocument()
    })

    it('creates existing as first', () => {
      render(
        <Provider>
          <New tpDays={['3']} edit setHasCloseButton={setHasCloseButton} />
        </Provider>
      )

      userEvent.click(screen.queryByText('Day'))
      userEvent.type(screen.getByRole('spinbutton'), '3')
      userEvent.click(screen.getByText('Save'))
      userEvent.click(screen.getByText('First'))

      expect(mockDispatch).toHaveBeenCalledWith(
        ACTIONS.TOUCHPOINT.EDIT_TOUCHPOINT({
          dayNum: 3,
          order: 'first'
        })
      )

      expect(setHasCloseButton).toHaveBeenCalled()
    })

    it('creates existing as last', () => {
      render(
        <Provider>
          <New tpDays={['3']} setHasCloseButton={setHasCloseButton} />
        </Provider>
      )

      userEvent.click(screen.queryByText('Day'))
      userEvent.type(screen.getByRole('spinbutton'), '3')
      userEvent.click(screen.getByText('Save'))
      userEvent.click(screen.getByText('Last'))

      expect(mockDispatch).toHaveBeenCalledWith(
        ACTIONS.TOUCHPOINT.NEW({
          dayNum: 3
        })
      )

      expect(setHasCloseButton).toHaveBeenCalledTimes(2)
    })
  })
})
