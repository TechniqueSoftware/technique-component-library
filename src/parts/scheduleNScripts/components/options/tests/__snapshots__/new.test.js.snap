// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<New /> should match snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="list-group list-group-flush"
    >
      <div
        class="border-0 d-flex flex-row fs-5 align-items-center list-group-item"
      >
        <div
          class="form-check form-check-inline"
        >
          <input
            class="form-check-input"
            name="immediate"
            type="radio"
          />
          <label
            class="form-check-label"
            title=""
          >
            Immediate
          </label>
        </div>
      </div>
      <div
        class="border-0 d-flex flex-row fs-5 align-items-center list-group-item"
      >
        <div
          class="form-check form-check-inline"
        >
          <input
            checked=""
            class="form-check-input"
            name="day"
            type="radio"
          />
          <label
            class="form-check-label"
            title=""
          >
            Day
          </label>
        </div>
        <div>
          <div
            class=""
          >
            <input
              class="form-control"
              id="day-touchpoint"
              min="1"
              name="day-touchpoint"
              required=""
              type="number"
              value="1"
            />
          </div>
        </div>
      </div>
      <div
        class="border-0 list-group-item"
      >
        <button
          class="
    d-flex
    align-items-center
    justify-content-center
    flex-row
    button
    
    ms-auto
   btn btn-primary btn-md"
          role="button"
          type="button"
        >
          <span>
            Save
          </span>
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
